src/apps/smallbank/app/smallbank.cpp

```diff
@@ -471,4 +472,9 @@ namespace ccfapp
   {
     return make_shared<SmallBank>(*nwt.tables, context);
   }
+
+  std::vector<ccf::js::FFIPlugin> get_js_plugins()
+  {
+    return {ccf::js::crypto_plugin};
+  }
```

Comment:
Can we put this in the weakly-linked default implementation, rather than repeating in every app?

Reply:
What about linking? Should the `ccf.{enclave,virtual}` targets always link to the plugin? If the plugin is not used (by overriding `get_js_plugins`, no code would be pulled in.

Reply:
Hmm. Maybe this default implementation should be in a separate file, so all the apps that want to use it are defined in CMake as `SRCS logging.cpp default_plugins.cpp`, so it looks like less boilerplate in the code? Still a breaking change for any consumer though. I'm tempted to aim really hard to avoid the breaking change (by making the default weak implementation include + require-linking-of these default modules). Ideally we're very lucky with the behaviour of weakly-linked symbols, and we can actually avoid linking their dependencies if they're actually overridden? But I assume that's not the case.

Reply:
> But I assume that's not the case.

Why not? Objects from static archives are only pulled in if they're referenced, which is true since we're not using `--whole-archive`.